### PR TITLE
fix(nextra): allow to contain dots in page filenames

### PR DESCRIPTION
**Pull request by B2o5T**
fixes https://github.com/shuding/nextra/issues/545

- [x] removes `getLocaleFromFilename`, `removeExtension` and `getFileName` in favor of simple and more generic `parseFileName` function
- [x] add tests for `parseFileName` to ensure it works like expected
- [x] fix `locales.ts`,  `const PUBLIC_FILE = /\.(.*)$/` not allow redirect page names with dots

### CODE REVIEW

File: packages/nextra/src/locales.ts [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
-
 export function locales(request: NextRequest) {
   const { nextUrl } = request
 
   const shouldHandleLocale =
-    !PUBLIC_FILE.test(nextUrl.pathname) &&
-    !nextUrl.pathname.includes('/api/') &&
-    !nextUrl.pathname.includes('/_next/') &&
+    !/^\/(api|_next)\//.test(nextUrl.pathname) &&
+    !/\.(jpe?g|svg|png|webmanifest)$/.test(nextUrl.pathname) &&
```
Comment (B2o5T) [inline packages/nextra/src/locales.ts:8]: https://github.com/vercel/next.js/issues/38699

File: packages/nextra/src/page-map.ts [hunk 2]
```diff
@@ -17,24 +15,24 @@ export function findPagesDir(dir: string = process.cwd()): string {
 
 export function getPageMap(
   currentResourcePath: string,
-  pageMaps: PageMapItem[],
+  pageMap: PageMapItem[],
```
Comment (B2o5T) [inline packages/nextra/src/page-map.ts:18]: typo

File: packages/nextra/src/utils.ts [hunk 3]
```diff
@@ -1,25 +1,26 @@
...
+  }
 }
 
-export const parseJsonFile: (
+export const parseJsonFile = (
   content: string,
   path: string
-) => Record<string, any> = (content: string, path: string) => {
+): Record<string, any> => {
   try {
```
Comment (B2o5T) [inline packages/nextra/src/utils.ts:24]: can be simplified

**Comment by shuding** [comment]
Great refactor, thanks!

**Pull request merged**